### ISSUE TITLE
Fetch token mint info

### ISSUE THREAD

**Issue by thegreat-vanderlinde** [vyper-protocol/vyper-otc-ui#110]
Currently we have hardcoded the reserve mint symbol:
https://github.com/vyper-protocol/vyper-otc-ui/blob/4e815ebd411676af2074f1aca27473a99e31090f/src/pages/contract/summary/%5Bid%5D.tsx#L47

We need to fetch token metadata using the [metaplex fungible token standard](https://docs.metaplex.com/programs/token-metadata/token-standard#the-fungible-standard).

We can use this snippet:
https://github.com/solana-labs/governance-ui/blob/6647cfc72fef5f418fd084feb809e1d2610f536d/components/treasuryV2/WalletList/WalletListItem/AssetList/index.tsx#L71-L93

**Comment by thegreat-vanderlinde** [comment]
@stevenhansel we can include the token info object in the `OtcState` object, this will help cause we'll need it multiple time in the future. you can add a new field close to the reserve mint info:

https://github.com/vyper-protocol/vyper-otc-ui/blob/8afc12b72eaa33f2003ef627ca47ac2a7659ef19/src/models/OtcState.ts#L15-L18

and you can fetch it here:
https://github.com/vyper-protocol/vyper-otc-ui/blob/8afc12b72eaa33f2003ef627ca47ac2a7659ef19/src/api/otc-state/fetchOtcState.ts#L25-L32

**Comment by stevenhansel** [comment]
Thanks for the information, that's a great idea to encapsulate the token info together inside the `OtcState`. I do have another question regarding fetching the token info data. 

```ts
const mintPubkey = new PublicKey(mintAddress)
const metadataAccount = findMetadataPda(mintPubkey)
const accountData = await connection.current.getAccountInfo(
  metadataAccount
)
```

```
import { SignerWalletAdapter } from '@solana/wallet-adapter-base'
...
current: SignerWalletAdapter | undefined
```

If you see the code from governance UI, it uses `SignerWalletAdapter` to get the account info. I'm not really sure, but I assume we have some similar adapter to get the account info. Can you give some leads regarding this @thegreat-vanderlinde , thanks

**Comment by thegreat-vanderlinde** [comment]
> I'm not really sure, but I assume we have some similar adapter to get the account info or do I have to install `@solana/wallet-adapter-base` and open the connection?

in order to fetch account info / data from chain you just need a valid `connection`. Since we're using [anchor](https://www.anchor-lang.com/) across the project we have the `AnchorProvider` that can be used. Consider it as a container for both the `wallet` (in case you need to sign some transaction, but that's not our case) and the `connection`.

For example here:
https://github.com/vyper-protocol/vyper-otc-ui/blob/8afc12b72eaa33f2003ef627ca47ac2a7659ef19/src/api/otc-state/fetchOtcState.ts#L27
we're using the `provider.connection`.


So, usually the process to fetch information from chain is to:
1. find the account we're interested in, for example with the method `findMetadataPda(mintPubkey)`
2. fetch the data stored in the account
3. decode those bytes in order to construct a meaningful object
We can surely stick with the points above, but considering that the metadata standard has been introduced by [metaplex](https://www.metaplex.com/) and [their sdks](https://github.com/metaplex-foundation/metaplex) are super well crafted, I'm quite sure they can offer some utility method that we can use.

**Comment by thegreat-vanderlinde** [comment]
Closed with #145
